### PR TITLE
Completed feature Book DTO

### PR DESCRIPTION
**Pull request by MinoDentori**
1. Added methods for BookController/Service/Repository (findAllBooks, getBookById, SaveBook).
2. Created DTOs for the Book class.

### CODE REVIEW

**Review by DevIgork — APPROVED** [review]
I think you did a great job

File: src/main/java/mate/academy/onlinebookstore/controller/BookController.java [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@RequiredArgsConstructor
+@RestController
+@RequestMapping(value = "/books")
```
Comment (bhdnchui) [inline src/main/java/mate/academy/onlinebookstore/controller/BookController.java]: Pay attention that url should contain `/api`.
`GET: /api/books`

File: src/main/java/mate/academy/onlinebookstore/model/Book.java [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+@Entity
+@Table(name = "book")
+public class Book {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+    private String title;
+    private String author;
+    @Column(unique = true)
+    private String isbn;
```
Comment (bhdnchui) [inline src/main/java/mate/academy/onlinebookstore/model/Book.java]: It is requirement from the first task, but please, add constraints to your model according to the task.
image

File: src/main/java/mate/academy/onlinebookstore/repository/BookRepository.java [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+
+import java.util.List;
+import mate.academy.onlinebookstore.model.Book;
+
+public interface BookRepository {
+    Book createBook(Book book);
+
+    List<Book> getAll();
+
+    Book getById(Long id);
```
Comment (bhdnchui) [inline src/main/java/mate/academy/onlinebookstore/repository/BookRepository.java]: Let's return `Optional<Book>` here.

**Review by bhdnchui — CHANGES_REQUESTED** [review]
Good job!
Please, pay attention to my comments. Also, take a look at this [doc](https://mate-academy.github.io/jv-program-fulltime/07_spring_boot/git-strategy) and it will help to handle several PRs in your repository in a right way.

File: src/main/java/mate/academy/onlinebookstore/controller/BookController.java (hunk 1 above)
Comment (MinoDentori) [reply src/main/java/mate/academy/onlinebookstore/controller/BookController.java]: fixed, I thought /api was like a placeholder for our domain.

File: src/main/java/mate/academy/onlinebookstore/model/Book.java (hunk 2 above)
Comment (MinoDentori) [reply src/main/java/mate/academy/onlinebookstore/model/Book.java]: fixed. @Column(nullable - false) used

File: src/main/java/mate/academy/onlinebookstore/repository/BookRepository.java (hunk 3 above)
Comment (MinoDentori) [reply src/main/java/mate/academy/onlinebookstore/repository/BookRepository.java]: fixed

**Comment by MinoDentori** [comment]
Fixed Comments. Tested via Postman.

**Review by vovasalyha — APPROVED** [review]

**Pull request merged**